### DISCUSSION TITLE
x-transition breaks with newlines

### DISCUSSION THREAD

**Discussion by mattvague** [alpinejs/alpine#4418] [7. Bugs (archived)]
I believe there's a bug in the current alpine transition logic regarding newlines. Currently if I have an `x-transition:leave` value like this in liquid (which renders out to having newlines):

```
  x-transition:leave="
    ease-out duration-[10000ms]
    pointer-events-none
  "
```

then an `InvalidCharacterError` error is thrown from `setClassesFromString` because Alpine is trying to treat the newlines characters as class names:

<img width="545" alt="image" src="https://github.com/user-attachments/assets/60404546-a7a1-4e26-8853-1d3c9cf7b91f">

HTML5 docs say [this](https://html.spec.whatwg.org/#classes) for `class` attribute:

> The attribute, if specified, must have a value that is a [set of space-separated tokens](http://www.w3.org/html/wg/drafts/html/master/single-page.html#set-of-space-separated-tokens) representing the various classes that the element belongs to.

And then says that:

>A set of space-separated tokens is a string containing zero or more words (known as tokens) separated by one or more [space characters](http://www.w3.org/html/wg/drafts/html/master/single-page.html#space-character), where words consist of any string of one or more characters, none of which are [space characters](http://www.w3.org/html/wg/drafts/html/master/single-page.html#space-character).

Where [whitespace characters](https://infra.spec.whatwg.org/#ascii-whitespace) are:

> U+0009 TAB, U+000A LF, U+000C FF, U+000D CR, or U+0020 SPACE.

I believe all we have to do to fix this is to change [this line](https://github.com/alpinejs/alpine/blob/7e636ec97d1eb5a1634274a5410e91abfe082874/packages/alpinejs/src/utils/classes.js#L15) as follows:

```diff
-    let split = classString => classString.split(' ').filter(Boolean)
+    let split = classString => classString.split(/\s/).filter(Boolean)
```

Happy to open a PR!

**Comment by ekwoka** [comment]
Makes sense!

**Reply by mattvague** [reply]
Sweet, will submit a PR

**Reply by mattvague** [reply]
Done! https://github.com/alpinejs/alpine/pull/4419
